% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/ti_mfa.R
\name{ti_mfa}
\alias{ti_mfa}
\title{MFA}
\usage{
ti_mfa(
  iter = 2000L,
  thin = 1L,
  pc_initialise = 1L,
  prop_collapse = 0L,
  scale_input = TRUE,
  zero_inflation = FALSE
)
}
\arguments{
\item{iter}{Number of MCMC iterations. Domain: U(20, 5000). Default: 2000.
Format: integer.}

\item{thin}{MCMC samples to thin. Domain: U(1, 20). Default: 1. Format:
integer.}

\item{pc_initialise}{Which principal component to initialise pseudotimes to.
Domain: U(1, 5). Default: 1. Format: integer.}

\item{prop_collapse}{Proportion of Gibbs samples which should marginalise over
c. Domain: U(0, 1). Default: 0. Format: numeric.}

\item{scale_input}{Logical. If true, input is scaled to have mean 0 variance 1.
Default: TRUE. Format: logical.}

\item{zero_inflation}{Logical, should zero inflation be enabled?. Default:
FALSE. Format: logical.}
}
\value{
A TI method wrapper to be used together with
\code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
}
\description{
Will generate a trajectory using
\href{https://doi.org/10.12688/wellcomeopenres.11087.1}{MFA}.

This method was wrapped inside a
\href{https://github.com/dynverse/ti_mfa}{container}.
The original code of this method is available
\href{https://github.com/kieranrcampbell/mfa}{here}.
}
\references{
Campbell, K.R., Yau, C., 2017. Probabilistic modeling of
bifurcations in single-cell gene expression data using a Bayesian mixture of
factor analyzers. Wellcome Open Research 2, 19.
}
\keyword{method}
