% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/ti_scuba.R
\name{ti_scuba}
\alias{ti_scuba}
\title{SCUBA}
\usage{
ti_scuba(
  rigorous_gap_stats = TRUE,
  N_dim = 2L,
  low_gene_threshold = 1L,
  low_gene_fraction_max = 0.7,
  min_split = 15L,
  min_percentage_split = 0.25
)
}
\arguments{
\item{rigorous_gap_stats}{Whether to use rigorous gap statistics to determine
number of clusters. Default: TRUE. Format: logical.}

\item{N_dim}{Number of TSNE dimensions. Domain: U(2, 3). Default: 2. Format:
integer.}

\item{low_gene_threshold}{Threshold value for genes of low expression levels.
Domain: U(0, 5). Default: 1. Format: numeric.}

\item{low_gene_fraction_max}{Maximum fraction of lowly-expressed cells allowed
for each gene. Domain: U(0, 1). Default: 0.7. Format: numeric.}

\item{min_split}{Lower threshold on the number of cells in a cluster for this
cluster to be split. Domain: U(1, 100). Default: 15. Format: integer.}

\item{min_percentage_split}{Minimum fraction of cells in the smaller cluster
during a bifurcation. Domain: U(0, 1). Default: 0.25. Format: numeric.}
}
\value{
A TI method wrapper to be used together with
\code{\link[dynwrap:infer_trajectories]{infer_trajectory}}
}
\description{
Will generate a trajectory using
\href{https://doi.org/10.1073/pnas.1408993111}{SCUBA}.

This method was wrapped inside a
\href{https://github.com/dynverse/ti_scuba}{container}.
The original code of this method is available
\href{https://github.com/gcyuan/SCUBA}{here}.
}
\references{
Marco, E., Karp, R.L., Guo, G., Robson, P., Hart, A.H., Trippa, L.,
Yuan, G.-C., 2014. Bifurcation analysis of single-cell gene expression data
reveals epigenetic landscape. Proceedings of the National Academy of Sciences
111, E5643â€“E5650.
}
\keyword{method}
